### ISSUE TITLE
Default Get-ChildItem / Get-Item output doesn't list the names of system-defined junctions on Windows

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#10630]
Related:
*  #10633 
* #7127

# Steps to reproduce

On Windows:

```powershell
Get-ChildItem $HOME -Force -Attributes "Directory+ReparsePoint" -Filter Cookies |
  Out-String | Should -match Cookies

Get-Item $HOME/Cookies -Force | Out-String | Should -match Cookies
```

Note the use of `Out-String` so as to expect the _default output formatting_, which is where the problem lies; by contrast, accessing the `Name` property explicitly, with, say, `Format-List Name`, works fine.

# Expected behavior

The tests should pass.

# Actual behavior

The tests fail, because the `Name` column in the default output is blank:

```none
Expected regular expression 'Cookies' to match '
    Directory: C:\Users\jdoe
Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
l--hs            6/5/2019 10:42 PM
', but it did not match.
```

# Environment data

```none
PowerShell Core 7.0.0-preview.4
```

**Comment by iSazonov** [comment]
I can not repo. It seems it was fixed in #7127.

**Comment by mklement0** [comment]
Looks like it, thanks.
